components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/util/JWTInboundAuthConfig.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.sso.jwt.util;
+
+import org.wso2.carbon.identity.application.common.model.InboundProvisioningConnector;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.mgt.AbstractInboundAuthenticatorConfig;
+
+/**
+ * This class is used for populating the GUI elements for the JWT SSO inbound authenticator in the Identity Server's
+ * SP configuration
+ *
+ * This class is extensible either by implementing the AbstractInboundAuthenticatorConfig or extending the
+ * JWTInboundAuthConfig. The customized class should be defined under
+ * <JWTSSO><AuthConfig></AuthConfig></JWTSSO> in identity.xml. Otherwise JWTInboundAuthConfig will be
+ * used as the default request processor.
+ */
+public class JWTInboundAuthConfig extends AbstractInboundAuthenticatorConfig
+        implements InboundProvisioningConnector {
+
+    private static final String NAME = "jwt-sso-inbound-auth";
+
+    public JWTInboundAuthConfig() {
+
+    }
+
+    @Override
+    public String getName() {
+
+        return NAME;
+    }
+
+    @Override
+    public String getConfigName() {
+
+        return NAME;
+    }
+
+    @Override
+    public String getFriendlyName() {
+        // The human-readable name that gets printed in the SP config.
+        return JWTInboundConstants.FRIENDLY_NAME;
+    }
+
+    @Override
+    public String getRelyingPartyKey() {
+
+        return JWTInboundConstants.SPBasedConfigs.RELYING_PARTY;
+    }
+
+    @Override
+    public Property[] getConfigurationProperties() {
+
+        Property relParty = new Property();
+        relParty.setName(JWTInboundConstants.SPBasedConfigs.RELYING_PARTY);
+        relParty.setDisplayName("Relying Party");
+        relParty.setRequired(true);
+        relParty.setDescription("The Relying Party ID to uniquely identify the service provider " +
```

Comment:
Wondering whether these strings should be localizable and whether we have a way to do it. Please check on that.

Reply:
Let's provide localization support for this from 5.11 dev portal onwards.